task4/apicode.py

```diff
@@ -0,0 +1,22 @@
+from flask import Flask, request, jsonify, session
+import redis
+
+app = Flask(__name__)
+
+# Set redis config
+cache = redis.StrictRedis(host='redis', port=6379, db=0)
+
+# Get Method, if key(name) not exist, return "Hello, Wiredcraft!"
+@app.route('/welcome', methods=["GET"])
+def Get_Name():
+    if(cache.get('name')):
+        return "Hello, " + cache.get('name').decode() + "!"
```

Comment:
any reason to ues `cache.get` twice in here?  that means for each GET api call it  calls redis twice

Reply:
Yes, there will be more additional waste, because of other things, I get distracted. I use variables to deal with this situation. It has been fixed.